### PR TITLE
verify file extensions

### LINKED ISSUE #199: File extension validation
atm the user can select any file to upload, even images. However, so far only CSV and soon Excel files are supported. Allowing only valid data types e.g. by setting the `accept` flag (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#Unique_file_type_specifiers) will avoid those kind of errors

image

### PR DESCRIPTION
**Pull request by sgratzl**
closes #199

### CODE REVIEW

**Comment by sgratzl** [comment]
Usually one implements that you are not allowed to drop the file in the first place by rejecting the drop. The file input already filters to show only valid files

**Comment by waxlamp** [comment]
Since I use the file input almost exclusively, I had to change the file filter options in order to test this branch. I admit this is a corner case, but all that means is that someone, somewhere will run into this.

I'm ok with not making the change on this branch, but in that case we should open an issue for it.

**Comment by sgratzl** [comment]
image

**Comment by subdavis** [comment]
I'm usually not a huge fan of toasts and snackbars for error display.  I think it dislocates the user action that caused the error and the error itself.

In this case I think it's OK (and possibly appropriate) because the upload type is already restricted, so users would have to change the rules of their file picker to even get to this error state.  There's also not really a better place to put this error.

File: web/src/components/Upload.vue [hunk 1]
```diff
@@ -69,19 +71,35 @@ export default {
...
-        progress: {},
-        meta: {},
-      })));
-
       const excelMimeTypes = [
         'application/vnd.ms-excel',
         'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
       ];
 
       const isExcelFile = file => excelMimeTypes.includes(file.type) || file.name.match(/\.xlsx?$/i);
```
Comment (subdavis) [inline web/src/components/Upload.vue]: This regex doesn't match the accepted types for the dropzone.  `.xlsx?` will match `.xls` too.

File: web/src/components/Upload.vue [hunk 2]
```diff
@@ -188,7 +210,8 @@ v-layout.upload-component(column, fill-height)
                   :items="dataTypes", label="Type of data",
                   item-text="name", item-value="value")
           v-divider(v-if="idx + 1 < files.length", :key="idx")
-    dropzone.filezone.mx-4.mb-4(:multiple="true", :message="message", @change="onFileChange")
+    dropzone.filezone.mx-4.mb-4(:multiple="true", :message="message", @change="onFileChange",
+        accept=".csv,.xlsx,.txt")
```
Comment (subdavis) [inline web/src/components/Upload.vue]: see above

File: web/src/components/Upload.vue [hunk 3]
```diff
@@ -69,19 +71,35 @@ export default {
...
-      })));
-
       const excelMimeTypes = [
         'application/vnd.ms-excel',
         'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
       ];
 
       const isExcelFile = file => excelMimeTypes.includes(file.type) || file.name.match(/\.xlsx?$/i);
+      const isCSVFile = file => file.type === 'text/csv' || file.name.match(/\.csv$/i);
+      const isTextFile = file => file.type === 'text/plain' || file.name.match(/\.txt$/i);
```
Comment (subdavis) [inline web/src/components/Upload.vue]: Could we move these out of the scope of this method, to the head of the script?

**Comment by waxlamp** [comment]
I agree with @subdavis's comments about toasts and snackbars etc.
